internal/pkg/pluginengine/change.go

```diff
@@ -169,3 +182,47 @@ func handleResult(smgr statemanager.Manager, change *Change) error {
 	log.Successf("Plugin %s(%s) %s done.", change.Tool.Name, change.Tool.Plugin.Kind, change.ActionName)
 	return nil
 }
+
+// FillInputParams fill inputs from state
+func fillRefValueWithOutputs(smgr statemanager.Manager, options map[string]interface{}) error {
+	// traverse options
+	for key, value := range options {
+		log.Debugf("Key: %s,  Value: %s.", key, value)
+		// judge whether the value is a string
+		if inst, ok := value.(string); ok {
+			// judge whether the format is ${{xxx}}
+			if strings.HasPrefix(inst, REF_PREFIX) && strings.HasSuffix(inst, REF_SUFFIX) {
+				ref := strings.TrimSpace(strings.TrimSuffix(strings.TrimPrefix(inst, REF_PREFIX), REF_SUFFIX))
```

Comment:
This train of trims makes it hard to read. Maybe we can create a function and give it a meaningful name.